specifyweb/frontend/js_src/lib/components/AppResources/TabDefinitions.tsx

```diff
@@ -97,29 +105,40 @@ const generateEditor = (xmlSpec: (() => BaseSpec<SimpleXmlNode>) | undefined) =>
     );
     const isReadOnly = React.useContext(ReadOnlyContext);
     return (
-      <CodeMirror
-        extensions={writable(extensions)}
-        readOnly={isReadOnly}
-        ref={handleRef}
-        theme={isDarkMode ? okaidia : xcodeLight}
-        onUpdate={({ state }): void => {
-          selectionRef.current = state.selection.toJSON();
-        }}
-        className={`w-full border border-brand-300 dark:border-none ${className}`}
-        /*
-         * Disable spell check if we are doing own validation as otherwise it's
-         * hard to differentiate between browser's spell check errors and our
-         * validation errors
-         */
-        spellCheck={typeof xmlSpec === 'function'}
-        value={data ?? ''}
-        /*
-         * FEATURE: provide supported attributes for autocomplete
-         *   https://codemirror.net/examples/autocompletion/
-         *   https://github.com/codemirror/lang-xml#api-reference
-         */
-        onChange={handleChange}
-      />
+      <div className="flex flex-col gap-2">
+        <div className="overflow-auto">
+          <CodeMirror
+            extensions={writable(extensions)}
+            readOnly={isReadOnly}
+            ref={handleRef}
+            theme={isDarkMode ? okaidia : xcodeLight}
+            onUpdate={({ state }): void => {
+              selectionRef.current = state.selection.toJSON();
+            }}
+            className={`w-full border border-brand-300 dark:border-none ${className}`}
+            /*
+             * Disable spell check if we are doing own validation as otherwise it's
+             * hard to differentiate between browser's spell check errors and our
+             * validation errors
+             */
+            spellCheck={typeof xmlSpec === 'function'}
+            value={data ?? ''}
+            /*
+             * FEATURE: provide supported attributes for autocomplete
+             *   https://codemirror.net/examples/autocompletion/
+             *   https://github.com/codemirror/lang-xml#api-reference
+             */
+            onChange={handleChange}
+          />
+        </div>
+        {showWarning &&
+          warnings.map((warning, index) => (
```

Comment:
see optional suggestion in https://github.com/specify/specify7/issues/4168#issuecomment-1783666322